exp/controllers/azureasomanagedcluster_controller.go

```diff
@@ -105,6 +127,33 @@ func (r *AzureASOManagedClusterReconciler) SetupWithManager(ctx context.Context,
 	return nil
 }
 
+func asoManagedControlPlaneToManagedClusterMap(c client.Client) handler.MapFunc {
```

Comment:
Should we check if the `ObjectMeta.DeletionTimestamp` is zero in order to not handle deleted AsoManagedControlPlanes?

Reply:
Is there a reason why we shouldn't be reconciling an AzureASOManagedCluster if an AzureASOManagedControlPlane is deleted?

Reply:
I'm not 100% sure but I noticed that the `AzureManagedControlPlaneToAzureManagedClusterMapper` does this and I'm wondering if the flow is different here.
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/ea3c0bd3719c6a55fd450843d8a77037900b729d/controllers/helpers.go#L905

Reply:
I'm not exactly sure what that's trying to accomplish, so I'd prefer not to add that here for now since I think it's a little safer to reconcile too much than not enough. Happy to add it later though if we find a compelling reason to.